### DISCUSSION TITLE
GitHub Token是必选项吗？

### DISCUSSION THREAD

**Discussion by flyweb291** [athul/waka-readme#123] [Q&A]
在使用工作流的时候，关于配置GH_TOKEN是必选项吗？

**Comment by flyweb291** [comment]
我的意思是如果我使用个人资料自述文件，就不需要提供GH_TOKEN了吗？不使用GH_TOKEN的话，他会不会权限不够？我现在对Action不太了解，为什么可以不需要GH_TOKEN就实现了文件的更新

**Comment by nandgator** [comment]
![image](https://github.com/athul/waka-readme/assets/38415384/d5ec23f1-7fbc-4675-b1df-2933cfe5e9f3)

Not unless you're using a non-profile repository.

![image](https://github.com/athul/waka-readme/assets/38415384/d8c6adff-09c1-4691-bc44-9f050d555f61)

We're pulling data from `https://wakatime.com/`.  Yes, you can provide you're own data as long as it aligns with [WakAPI](https://github.com/muety/wakapi). And no, it does not require GH_Token if you're using you're profile [readme](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-github-profile/customizing-your-profile/managing-your-profile-readme).
